### ISSUE TITLE
Dragging a class to change superclass gets nil DNU #notEmpty

### ISSUE THREAD

**Issue by daniels220** [dolphinsmalltalk/Dolphin#271]
`#moveClass:toSuperclass:` creates a ClassBuilder without setting `instanceVariables`. `ClassBuilder>>setInstanceVariablesOf:` used to contain a nil-check, but this disappeared when switching to array storage of inst vars and creating `ClassDescription>>setInstanceVariables:`. It _does_ optimize for the no-inst-vars case by storing nil instead of an empty array, but this gets an error if the argument is _already_ nil. Note, the old and new superclass must have different inst vars (`#instanceVariablesAreBeingChanged` must answer true) for the problem to occur.

Does it seem reasonable to just add a nil-check to `#setInstanceVariables:`, e.g. `instanceVariables := (aCollection isNil or: [aCollection isEmpty]) ifFalse: [aCollection]`? Or should the nil-check be in `#setInstanceVariablesOf:` as before, and avoid calling `#setInstanceVariables:` at all if the argument would be nil? (This is how it was in D6, and seems strange, but I'm not clear on the context and possible implications...)

**Comment by blairmcg** [comment]
Wouldn't the best fix here be:

```
ClassBuilder>>setInstanceVariablesOf: aClass
...
    aClass setInstanceVariablesOf: **self** instanceVariables
```

?

Again there is a test hole though.

**Comment by daniels220** [comment]
D'oh! I don't know how I didn't see that method. I used that fix and added tests—see pull #273.
